File: img/FalaMuriloAndroid/app/src/main/java/com/tairone/falamurilo/models/CardItem.kt
package com.tairone.falamurilo.models

import androidx.annotation.DrawableRes
import androidx.compose.ui.graphics.Color
import com.tairone.falamurilo.R

data class CardItem(
    val id: String,
    @DrawableRes val imageRes: Int,
    val labels: Map<String, String>,
    val color: Color,
) {
    fun label(language: AppLanguage): String =
        labels[language.code] ?: labels["en-US"] ?: id

    companion object {
        val all: List<CardItem> by lazy {
            listOf(
                CardItem("eat",      R.drawable.card_comer,    mapOf("en-US" to "Eat",      "es-ES" to "Comer",    "pt-BR" to "Comer"),    Color(0xFFFF6B6B)),
                CardItem("drink",    R.drawable.card_beber,    mapOf("en-US" to "Drink",    "es-ES" to "Beber",    "pt-BR" to "Beber"),    Color(0xFF4ECDC4)),
                CardItem("play",     R.drawable.card_brincar,  mapOf("en-US" to "Play",     "es-ES" to "Jugar",    "pt-BR" to "Brincar"),  Color(0xFF45B7D1)),
                CardItem("bathroom", R.drawable.card_banheiro, mapOf("en-US" to "Bathroom", "es-ES" to "Baño",     "pt-BR" to "Banheiro"), Color(0xFF96CEB4)),
                CardItem("dad",      R.drawable.card_pai,      mapOf("en-US" to "Dad",      "es-ES" to "Papá",     "pt-BR" to "Papai"),    Color(0xFFFFEAA7)),
                CardItem("mom",      R.drawable.card_mae,      mapOf("en-US" to "Mom",      "es-ES" to "Mamá",     "pt-BR" to "Mamãe"),    Color(0xFFDDA0DD)),
                CardItem("yes",      R.drawable.card_sim,      mapOf("en-US" to "Yes",      "es-ES" to "Sí",       "pt-BR" to "Sim"),      Color(0xFF98D8C8)),
                CardItem("no",       R.drawable.card_nao,      mapOf("en-US" to "No",       "es-ES" to "No",       "pt-BR" to "Não"),      Color(0xFFF7A8A8)),
                CardItem("help",     R.drawable.card_ajuda,    mapOf("en-US" to "Help",     "es-ES" to "Ayuda",    "pt-BR" to "Ajuda"),    Color(0xFFFFD700)),
            )
        }
    }
}

File: FalaMurilo/FalaMuriloIOS/Models/CardItem.swift
import SwiftUI

struct CardItem: Identifiable, Hashable {
    let id: String
    let imageName: String          // asset catalog name
    let labels: [String: String]
    let color: Color

    static let all: [CardItem] = [
        CardItem(id: "eat",      imageName: "comer",    labels: ["en-US": "Eat",      "es-ES": "Comer",    "pt-BR": "Comer"],    color: Color(hex: "FF6B6B")),
        CardItem(id: "drink",    imageName: "beber",    labels: ["en-US": "Drink",    "es-ES": "Beber",    "pt-BR": "Beber"],    color: Color(hex: "4ECDC4")),
        CardItem(id: "play",     imageName: "brincar",  labels: ["en-US": "Play",     "es-ES": "Jugar",    "pt-BR": "Brincar"],  color: Color(hex: "45B7D1")),
        CardItem(id: "bathroom", imageName: "banheiro", labels: ["en-US": "Bathroom", "es-ES": "Baño",     "pt-BR": "Banheiro"], color: Color(hex: "96CEB4")),
        CardItem(id: "dad",      imageName: "pai",      labels: ["en-US": "Dad",      "es-ES": "Papá",     "pt-BR": "Papai"],    color: Color(hex: "FFEAA7")),
        CardItem(id: "mom",      imageName: "mae",      labels: ["en-US": "Mom",      "es-ES": "Mamá",     "pt-BR": "Mamãe"],    color: Color(hex: "DDA0DD")),
        CardItem(id: "yes",      imageName: "sim",      labels: ["en-US": "Yes",      "es-ES": "Sí",       "pt-BR": "Sim"],      color: Color(hex: "98D8C8")),
        CardItem(id: "no",       imageName: "nao",      labels: ["en-US": "No",       "es-ES": "No",       "pt-BR": "Não"],      color: Color(hex: "F7A8A8")),
        CardItem(id: "help",     imageName: "ajuda",    labels: ["en-US": "Help",     "es-ES": "Ayuda",    "pt-BR": "Ajuda"],    color: Color(hex: "FFD700")),
    ]
}

extension Color {
    init(hex: String) {
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let r = Double((int >> 16) & 0xFF) / 255
        let g = Double((int >> 8)  & 0xFF) / 255
        let b = Double( int        & 0xFF) / 255
        self.init(.sRGB, red: r, green: g, blue: b, opacity: 1)
    }
}
